lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Oh cool, thanks @dtzWill !  I was offline for a while today so I didn't get a chance to look at the commit/discussion log.  

 I'll dig in a bit more here..  If I get stuck I'll ask around or post to the mailing list in a day or so...